cfme/services/catalogs/catalog_items/__init__.py

```diff
@@ -294,18 +318,47 @@ def delete(self):
         assert view.is_displayed
         view.flash.assert_success_message('The selected Catalog Item was deleted')
 
-    def add_button_group(self):
-        button_name = fauxfactory.gen_alpha()
-        view = navigate_to(self, 'AddButtonGroup')
-        view.fill({'btn_group_text': 'group_text',
-                   'btn_group_hvr_text': button_name,
-                   'btn_image': self.button_icon_name})
+    def add_button_group(self, **kwargs):
+        button_name = kwargs.get("text", "gp_{}".format(fauxfactory.gen_alpha()))
+        hover_name = kwargs.get("hover", "hover_{}".format(fauxfactory.gen_alpha()))
+        image = kwargs.get("image", "broom")
+
+        view = navigate_to(self, "AddButtonGroup")
+        view.fill(
+            {
+                "btn_group_text": button_name,
+                "btn_group_hvr_text": hover_name,
+                "btn_image": image,
+            }
+        )
         view.add.click()
-        view = self.create_view(DetailsCatalogItemView)
-        assert view.is_displayed
+        view = self.create_view(DetailsCatalogItemView, wait="10s")
         view.flash.assert_no_error()
         return button_name
 
+    def button_group_exists(self, name):
+        view = navigate_to(self, "Details")
+        path = view.catalog_items.tree.read()
+        path.extend(["Actions", "{} (Group)".format(name)])
+        try:
+            view.catalog_items.tree.fill(path)
+            return True
+        except Exception as e:
+            logger.exception(e)
```

Comment:
The exception is logged automatically when using logger.exception, you should pass a contextual message for the exception here. Help the reader of the log understand what was being done when this exception occurred. 

Also, I would prefer to have targeted exception catching here, instead of a diaper - is there some expected set of exceptions that might occur when trying to fill the tree? CandidateNotFound? ItemNotFound?

Reply:
I use this finally when I was unable to find exception but still I would like to find exact exception.